scala-thoughts.md

```diff
@@ -0,0 +1,5 @@
+I really like scala so far. So far it seems like it gives me everything I liked about functional Python and Haskell with the benefits of a static type system over Python and flexibility over Haskell. Parts of it remind me of Java and I feel like they're mostly vestiges from it being made in Java, and they don't really affect the feel fo the language very much. I actually don't really have any qualms at all right now; it seems like a good language. 
+
+I have plenty of things that I'm not sure about. What's the difference between a seq, and a list, and a type with an asterisk at the end? They all seem to function the same, i.e. like a Python list, at least for the first two. I don't feel like I ever treated seqs and lists differently throughout this assignment even though they were both used a lot. I'm still not sure about style conventions within Scala and I'm not sure if I've been following them or horribly mangling them. I'm not sure to what degree I should be writing one-liners (which I ~~probably~~ definitely did too much but I like writing one-liners). Comparing mine to the lab's solutions further confuses this because they did plenty of things that I think could have been done better (like checkValuesIncrease in recursionpatternmatchingexercise) or that I have no idea how they work (like compress, which looks like it would turn (1,1,2,3,1,1) into (1,2,3,1) rather than (1,2,3)). This also goes for Ben's solution. That said looking at both solutions has given me a better understanding of some of the built-in functions like span and partition and when they might be useful.
+
+There are also plenty of things that I think still just haven't quite clicked, like the OOP system which seems much much better than Java's but still a bit weird. Companion objects are weird. Some bits of syntax like function declarations are weird (but not necessarily in a bad way). I'm also a bit unsure about tools like sbt and eclipse. It seems like ScalaIDE seems to arbitrarily catch type errors that don't actually exist, and then refuse to compile my code until I delete and retype stuff until they go away, which is a bit annoying. Overall it's pretty helpful though. Unit testing was always Eclipse's strong point. Scala's tests are really weird, though. 
```

Comment:
_Sooo_ many OO things, which we'll start talking about and taking apart this week.